components/core/src/service.rs

```diff
@@ -482,8 +485,16 @@ mod test {
     }
 
     #[test]
-    fn service_group_from_str_too_many_periods() {
-        let group = "only.one.period@allowed";
+    fn service_group_from_str_with_period_in_service() {
```

Comment:
Even though we're really not using the application, environment, and organization bits of the service group name right now, It would be nice to have some tests that have a service group name with all of them, as well as a service name that has dots, just to ensure that everything stays consistent.